pkg/asset/fsck.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package asset
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// detectFileSystemType uses blkid to identify the file system type of a given image file
+func detectFileSystemType(imagePath string) (string, error) {
+	cmd := exec.Command("blkid", "-o", "value", "-s", "TYPE", imagePath)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to detect file system type: %v", err)
+	}
+	return string(output), nil
+}
+
+// runFsck uses the fs-specific fsck subcommand to check if an image file is clean
+func runFsck(imagePath string, fsType string) (bool, error) {
+	fsckCommand := fmt.Sprintf("fsck.%s", fsType)
+	cmd := exec.Command(fsckCommand, "-n", imagePath)
+	cmd.Run()
+
+	exitCode := cmd.ProcessState.ExitCode()
+	return exitCode == 0, nil
+}
+
+// Fsck returns whether a filesystem is clean or not
+func Fsck(r io.Reader) (bool, error) {
```

Comment:
Revert the order of functions in this file so that this one comes first. It's the entry point and the most important one.